tests/pipelines/test_pipelines_common.py

```diff
@@ -2395,40 +2320,62 @@ def test_pipeline_level_group_offloading_sanity_checks(self):
                 self.assertTrue(torch.device(component_device).type == torch.device(offload_device).type)
 
     @require_torch_accelerator
-    def test_pipeline_level_group_offloading_inference(self, expected_max_difference=1e-4):
-        components = self.get_dummy_components()
-        pipe: DiffusionPipeline = self.pipeline_class(**components)
+    def test_group_offloading_inference(self):
+        if not self.test_group_offloading:
+            pytest.skip("`test_group_offloading` is disabled hence skipping.")
 
-        for name, component in pipe.components.items():
-            if hasattr(component, "_supports_group_offloading"):
-                if not component._supports_group_offloading:
-                    pytest.skip(f"{self.pipeline_class.__name__} is not suitable for this test.")
+        def create_pipe():
+            torch.manual_seed(0)
+            components = self.get_dummy_components()
+            pipe = self.pipeline_class(**components)
+            pipe.set_progress_bar_config(disable=None)
+            return pipe
 
-        # Regular inference.
-        pipe = pipe.to(torch_device)
-        pipe.set_progress_bar_config(disable=None)
-        torch.manual_seed(0)
-        inputs = self.get_dummy_inputs(torch_device)
-        inputs["generator"] = torch.manual_seed(0)
-        out = pipe(**inputs)[0]
+        def enable_group_offload_on_component(pipe, group_offloading_kwargs):
+            # We intentionally don't test VAE's here. This is because some tests enable tiling on the VAE. If
+            # tiling is enabled and a forward pass is run, when accelerator streams are used, the execution order of
+            # the layers is not traced correctly. This causes errors. For apply group offloading to VAE, a
+            # warmup forward pass (even with dummy small inputs) is recommended.
+            exclude_modules = []
```

Comment:
Can't we just set directly? Why nested loops? Or even set them via  `group_offloading_kwargs`?
```py
exclude_modules = ["vae", "vqvae", "image_encoder"]
```